chore: update project to Python 3.10

jvirgovic opened the pull request:
Code can now succesfully run on python version 3.10. Tested all functions (web, cli...)
- got rid of useless stuff from .gitignore, which is not gonna be generated using grapheneX
- update shebang to use python 3.10
- updated pyproject.toml
- generated new poetry.lock

This MR doesn't include the newest versions of dependencies which can run with Python version 3.10 as I am still testing them.
There are a lot of issues with flask-socketio, and basically with flask in general, and the best solution I've found so far is: "just downgrade"...

Commit:
v3.10

jvirgovic commented:
This will also get rid of all ImportErrors except for the "from collections import mapping", which is easily fixable than all the other ones I had to deal with when building from src

orhun commented:
Is that possible to get rid of "from collections import mapping" error as well?

Also, can you update README.md accordingly to these changes if needed?

jvirgovic commented:
Yes, I'll update README.

Regarding the collections ImportError: This is caused by some other dependency in code, which is outdated.
I tried running grapheneX with most up-to-date dependencies, but as you can expect some parts of the project stopped working completely, so this will need a little more time & testing.

I'd also like to get rid of:
`# -*- coding: utf-8 -*-`
As python3 already has utf-8 as default encoding, and we are not aiming for python2 compatibility.

Commit:
PEP-8

jvirgovic commented:
- Updated codebase according to PEP-8
- Changed python version in README.md
- Got rid of source encoding header, which is no longer needed
- Changed some class methods to static ones

jvirgovic commented:
@orhun any thoughts?

orhun commented on graphenex/core/web/__init__.py:
```diff
@@ -1,11 +1,13 @@
 
+from graphenex.core.cli.shell import Shell
+from graphenex.core.utils.logcl import GraphenexLogger
+from graphenex.core.web.views import *
+from graphenex.core.web.providers import *
```
You should import these after initializing the logger etc. otherwise we get this error:

```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/lib/python3.11/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1147, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/home/orhun/gh/grapheneX/graphenex/__main__.py", line 6, in <module>
    from graphenex.core.web import run_server
  File "/home/orhun/gh/grapheneX/graphenex/core/web/__init__.py", line 6, in <module>
    from graphenex.core.web.views import *
  File "/home/orhun/gh/grapheneX/graphenex/core/web/views.py", line 10, in <module>
    from graphenex.core.web import app, logger, socketio
ImportError: cannot import name 'app' from partially initialized module 'graphenex.core.web' (most likely due to a circular import) (/home/orhun/gh/grapheneX/graphenex/core/web/__init__.py)
```

Patch: 

```diff
diff --git a/graphenex/core/web/__init__.py b/graphenex/core/web/__init__.py
index 0b90ddb..5dd1eb0 100644
--- a/graphenex/core/web/__init__.py
+++ b/graphenex/core/web/__init__.py
@@ -6,9 +6,6 @@ from flask_socketio import SocketIO
 
 from graphenex.core.cli.shell import Shell
 from graphenex.core.utils.logcl import GraphenexLogger
-from graphenex.core.web.views import *
-from graphenex.core.web.providers import *
-
 
 logger = GraphenexLogger(__name__)
 app = Flask(__name__)
@@ -17,6 +14,8 @@ app.config['ACCESS_TOKEN'] = secrets.token_urlsafe(6)
 socketio = SocketIO(app)
 default_addr = ('localhost', '8080')
 
+from graphenex.core.web.views import *
+from graphenex.core.web.providers import *
 
 def disable_flask_logs():
     """Disable the Flask logs"""

```

jvirgovic replied:
My bad, fixed this

orhun commented:
This is great, thanks a lot! 🐻 Just had one request for fixing the build error.

Commit:
Move imports back

jvirgovic commented:
Should be ready to merge :)

orhun commented on graphenex/core/web/__init__.py:
```diff
@@ -1,11 +1,13 @@
 
+from graphenex.core.cli.shell import Shell
+from graphenex.core.utils.logcl import GraphenexLogger
+from graphenex.core.web.views import *
+from graphenex.core.web.providers import *
```
Please remove duplicate imports :)

jvirgovic replied:
Oof, how did i miss this again..Should be resolved now

Commit:
remove duplicate imports

orhun reviewed (approved):
💯

orhun commented:
Are you planning to submit more PRs, should we cut a release soon?

jvirgovic commented:
I'd like to make some changes to descriptions of hardening modules, and some other small stuff, and than we can release a new version.

I'd also like to tackle #136, #131, but those will take more time, and can be added in later version

orhun commented:
Sounds good! Let me know when you think everything is ready for a release. I will also have to update the [AUR package](https://aur.archlinux.org/packages/graphenex) afterwards...
